core/templates/dev/head/services/UpgradedServices.ts

```diff
@@ -32,9 +34,9 @@ export class UpgradedServices {
     'UtilsService': new UtilsService(),
     'SidebarStatusService': new SidebarStatusService(
       new WindowDimensionsService()),
-    'WindowDimensionsService': new WindowDimensionsService()
+    'WindowDimensionsService': new WindowDimensionsService(),
+    'UrlService': new UrlService(new WindowRef())
```

Comment:
Should `WindowRef` be part of this list?

Reply:
@bansalnitish what are your thoughts on it? I am not sure about it either.

Reply:
And I believe we only need to add those services here which are being used by our code directly for the e2e tests.